### PR TITLE
[sophos_central] Initial Release for the Sophos Central

### PR DESCRIPTION
**Pull request by brijesh-elastic**
## What does this PR do?

- Generated the skeleton of the Sophos Central integration package.
- Added data streams.
- Added data collection logic for all the data streams.
- Added the ingest pipeline for all the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for all the data streams.
- Added system test cases for all the data streams.

## Integration release checklist
This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target is documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^7.17.0 || ^8.0.0

### New Package
- [x] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes
- [x] Dashboards exists
- [x] Screenshots added or updated
- [x] Datastream filters added to visualizations

### Log dataset changes
- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists
- [x] Sample event (sample_event.json) exists

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/sophos_central directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes https://github.com/elastic/integrations/issues/1149

## Screenshots

image
image
image
image
image
image

### CODE REVIEW

File: packages/sophos_central/data_stream/alert/fields/base-fields.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+- name: data_stream.type
```
[the remark this answers is not in the document]
Comment (brijesh-elastic) [reply packages/sophos_central/data_stream/alert/fields/base-fields.yml:1]: Hi @efd6 
We have been following best practice by not converting the data-stream and event fields of base-fields.yml to ecs fields. Also, base-fields.yml is an automatically generated file. Let me know if we need to make the changes.

[the remark this answers is not in the document]
Comment (efd6) [reply packages/sophos_central/data_stream/alert/fields/base-fields.yml:1]: I'm happy for this to be left until we resolve what should be done across all packages.

**Review by efd6 — APPROVED** [review]
Thanks

File: packages/sophos_central/data_stream/alert/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+      ignore_missing: true
+      on_failure:
+        - append:
+            field: error.message
+            value: '{{{_ingest.on_failure_message}}}'
+  - rename:
+      field: json.data.app_id
+      target_field: sophos_central.alert.data.app_id
+      ignore_missing: true
+  - append:
```
Comment (P1llus) [inline packages/sophos_central/data_stream/alert/elasticsearch/ingest_pipeline/default.yml:601]: Do we really need this twice? Seeing as we catch it in `on_failure:` as well?

**Review by P1llus — APPROVED** [review]

**Pull request merged**